### PR TITLE
Charlesmchen/arbitrary attachments

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

**Comment by charlesmchen** [comment]
See: https://github.com/WhisperSystems/SignalServiceKit/pull/174

File: Signal/src/ViewControllers/SignalAttachment.swift [hunk 1]
```diff
@@ -249,13 +216,7 @@ class SignalAttachment: NSObject {
         guard UIPasteboard.general.numberOfItems >= 1 else {
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/SignalAttachment.swift]: nit: a little simpler if this whole method is replaced by:

```
return UIPasteboard.general.numberOfItems > 0
```

File: Signal/src/ViewControllers/SignalAttachment.swift [hunk 2]
```diff
@@ -300,9 +261,13 @@ class SignalAttachment: NSObject {
                 return audioAttachment(data : data, dataUTI : dataUTI)
             }
         }
-        // TODO: We could handle generic attachments at this point.
 
-        return nil
+        let dataUTI = pasteboardUTISet[pasteboardUTISet.startIndex]
+        guard let data = dataForFirstPasteboardItem(dataUTI:dataUTI) else {
+            Logger.verbose("\(TAG) Missing expected pasteboard data for UTI: \(dataUTI)")
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/SignalAttachment.swift]: nit: This *seems* like an error condition? 

If so, I'd say a Logger.error and assertionFailure are in order.

File: Signal/src/ViewControllers/SignalAttachment.swift (hunk 1 above)
Comment (charlesmchen) [inline Signal/src/ViewControllers/SignalAttachment.swift]: Done.

File: Signal/src/ViewControllers/SignalAttachment.swift (hunk 2 above)
Comment (charlesmchen) [inline Signal/src/ViewControllers/SignalAttachment.swift]: Done.

**Pull request merged**